testing/trino-tests/src/test/java/io/trino/security/TestAccessControl.java

```diff
@@ -756,4 +818,112 @@ public void testUseStatementAccessControlWithDeniedSchema()
+                "Cannot access properties: \\[string_materialized_view_property]");
+        assertAccessDenied(
+                "ALTER MATERIALIZED VIEW mock.default.test_materialized_view SET PROPERTIES string_materialized_view_property = 'value'",
+                "Cannot access properties: \\[string_materialized_view_property]");
+        assertAccessDenied(
+                "ALTER MATERIALIZED VIEW mock.default.test_materialized_view SET PROPERTIES STRING_MATERIALIZED_VIEW_PROPERTY = 'value'",
+                "Cannot access properties: \\[string_materialized_view_property]");
+        assertAccessDenied(
+                "ALTER MATERIALIZED VIEW mock.default.test_materialized_view SET PROPERTIES string_materialized_view_property = DEFAULT",
+                "Cannot access properties: \\[string_materialized_view_property]");
+    }
+
+    @Test
+    public void testPropertiesAccessControlIsSkippedWhenUsingDefaults()
+    {
+        systemAccessControl.set(new DenySetPropertiesSystemAccessControl());
+        systemAccessControl.set(new DenySetPropertiesSystemAccessControl());
+        assertAccessAllowed("CREATE TABLE mock.default.new_table (pk bigint)");
+        assertAccessAllowed("CREATE SCHEMA mock.new_schema");
+        assertAccessAllowed("CREATE MATERIALIZED VIEW mock.default.new_materialized_view AS SELECT 1 a");
+    }
+
+    private static final class DenySetPropertiesSystemAccessControl
+            extends AllowAllSystemAccessControl
+    {
+        @Override
+        public void checkCanCreateSchema(SystemSecurityContext context, CatalogSchemaName schema, Map<String, Object> properties)
+        {
+            checkProperties(properties);
+        }
+
+        @Override
+        public void checkCanCreateTable(SystemSecurityContext context, CatalogSchemaTableName table, Map<String, Object> properties)
+        {
+            checkProperties(properties);
+        }
+
+        @Override
+        public void checkCanCreateMaterializedView(SystemSecurityContext context, CatalogSchemaTableName materializedView, Map<String, Object> properties)
+        {
+            checkProperties(properties);
+        }
+
+        @Override
+        public void checkCanSetTableProperties(SystemSecurityContext context, CatalogSchemaTableName table, Map<String, Optional<Object>> properties)
+        {
+            checkProperties(properties);
+        }
+
+        @Override
+        public void checkCanSetMaterializedViewProperties(SystemSecurityContext context, CatalogSchemaTableName materializedView, Map<String, Optional<Object>> properties)
+        {
+            checkProperties(properties);
+        }
+
+        private static void checkProperties(Map<?, ?> properties)
+        {
+            if (!properties.isEmpty()) {
+                throw new AccessDeniedException("Cannot access properties: " + properties.keySet());
```

Comment:
do you have any test that would try to set 2+ properties?

i think it would be nice to have
but then, use of keySet could cause test flakiness (indeterminism), unless we eg sort